lib/spack/spack/installer.py

```diff
@@ -2466,7 +2707,11 @@ def install(self):
         """
         try:
             with fs.replace_directory_transaction(self.task.pkg.prefix):
-                self.installer._install_task(self.task, self.install_status)
+                rc = self.installer._install_task(self.task, self.install_status)
```

Comment:
🤔 "Reaching into" the installer's private method makes me wonder if there's a better design option here.

Reply:
Not really related to this, but yes I agree -- I'll see if I can handle this at the same time.